artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -462,7 +285,12 @@ public void cleanup() {
                // Then we do some check on eventual pages that can be already removed but they are away from the streaming
                cleanupMiddleStream(depagedPages, depagedPagesSet, cursorList, minPage, firstPage);
 
-               if (pagingStore.getNumberOfPages() == 0 || pagingStore.getNumberOfPages() == 1 && pagingStore.getCurrentPage().getNumberOfMessages() == 0) {
+               if (pagingStore.getNumberOfPages() < 0) {
+                  new Exception("WHAT???").printStackTrace(System.out);
```

Comment:
needs tidied up

Reply:
OOps... thanks...  Debug message!

Reply:
I add a "//TODO: delete/update/etc" style comment whenever I add 'hacky working debug' stuff like this, makes it easy to find later, especially in a large diff.